projects/slap_manipulation/scripts/slap_inference.py

```diff
@@ -0,0 +1,124 @@
+import hydra
+import numpy as np
+import rospy
+import torch
+import trimesh
+from slap_manipulation.env.stretch_manipulation_env import StretchManipulationEnv
+from slap_manipulation.policy.action_prediction_module import APModule
+from slap_manipulation.policy.interaction_prediction_module import IPModule
+
+from home_robot.utils.point_cloud import show_point_cloud
+
+# TODO: move this to a constants file
+STRETCH_GRIPPER_MAX = 0.6
+
+
+def create_apm_input(raw_data, p_i):
+    raise NotImplementedError
+
+
+def create_ipm_input(
+    raw_data: dict, lang: list[str], filter_depth=False, debug=False, num_pts=8000
+):
+    input_vector = ()
+    depth = raw_data["depth"]
+    rgb = raw_data["rgb"].astype(np.float64)
+    xyz = raw_data["xyz"].astype(np.float64)
+    camera_pose = raw_data["camera_pose"]
+    proprio = raw_data["gripper_state"].astype(np.float64)
+    if proprio < 0.8 * STRETCH_GRIPPER_MAX:
+        proprio = np.array([1.0, proprio, -1.0])
+    else:
+        proprio = np.array([0.0, proprio, -1.0])
+
+    depth = depth.reshape(-1)
+    rgb = rgb.reshape(-1, 3)
+    cam_xyz = xyz.reshape(-1, 3)
+    xyz = trimesh.transform_points(cam_xyz, camera_pose)
+
+    # apply depth and z-filter for comparative distribution to training data
+    if filter_depth:
```

Comment:
when would we not want this? is this default?

Reply:
This is just for testing right now. Since I was going to z-filter Stretch inputs to match table-top environment.